rest/replicator_test.go

```diff
@@ -3810,3 +3810,236 @@ func TestActiveReplicatorPullConflictReadWriteIntlProps(t *testing.T) {
 		})
 	}
 }
+
+// This test ensures that the tombstone revision wins over non-tombstone revision
+// whilst applying default conflict resolution policy through pushAndPull replication.
+func TestDefaultConflictResolverWithTombstone(t *testing.T) {
+	if base.GTestBucketPool.NumUsableBuckets() < 2 {
+		t.Skipf("test requires at least 2 usable test buckets")
+	}
+	if !base.TestUseXattrs() {
+		t.Skip("This test only works with XATTRS enabled")
+	}
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+
+	// startReplication starts the replication tasks and asserts that the start up
+	// operation is successful. It returns a teardown function to stop the replication
+	// that can be called in defer.
+	startReplication := func(ar *db.ActiveReplicator) func() {
+		require.NoError(t, ar.Start(), "Error starting replication")
+		return func() { require.NoError(t, ar.Stop(), "Error stopping replication") }
+	}
+
+	// requireBodyEmpty asserts that the specified document body is empty.
+	requireBodyEmpty := func(rt *RestTester, docID string) {
+		var body []byte
+		_, err := rt.Bucket().Get(docID, &body)
+		require.EqualError(t, err, fmt.Sprintf("Error during Get %s: key not found", docID))
```

Comment:
This doesn't look like it's actually requiring an empty body - it's requiring a key not found error.  Should be renamed for clarity.